### DISCUSSION TITLE
What is batect's equivalent of `docker build --output type=local`?

### DISCUSSION THREAD

**Discussion by davidtan-tw** [batect/batect#1403] [Q&A] [answered]
Hi folks,

I'm using batect in a Python project (https://github.com/davidtan-tw/poetry-spike). I'm using Poetry as a dependency manager, and Dockerfile, I run `poetry install` (like `npm install`) and it creates a lockfile (`poetry.lock`, like `package-lock.json`) which Poetry maintainers recommend should be checked into the repo.

However, the problem is that `poetry.lock` is "trapped" inside the docker image, and is overwritten when I start my dev container and mount my project directory into workdir.

How do people typically solve this problem? Should I be looking for an equivalent of [--output type=local](https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs) in batect?

**Comment by charleskorn** [comment, marked as answer]
Hey @davidtan-tw, thanks for the question. Typically, I wouldn't try to retrieve a file from the image build process like this. In other similar situations, I've created a `setup` task that downloads dependencies etc. (eg. [the TypeScript sample project](https://github.com/batect/batect-sample-typescript/blob/main/batect.yml#L26)). 

Would that work for your situation? I'm not familiar with Poetry, so I can't comment too much on that.

**Reply by davidtan-tw** [reply]
Hi @charleskorn, that's a great solution. I've done the same for my Python project, and after running `./batect setup` I can access `poetry.lock` (equivalent of `package-lock.json` or `yarn.lock`) on the filesystem and check it into git. 

Thanks so much!
